Plugins/NativeEngine/Source/NativeEngine.cpp

```diff
@@ -2261,6 +2263,16 @@ namespace Babylon
         }
     }
 
+    void NativeEngine::PopulateFrameStats(const Napi::CallbackInfo& info)
+    {
+        const auto updateToken{m_update.GetUpdateToken()};
+        const auto stats{bgfx::getStats()};
+        const double toGpuNs = 1000000000.0 / double(stats->gpuTimerFreq);
+        const double gpuTimeNs = (stats->gpuTimeEnd - stats->gpuTimeBegin) * toGpuNs;
+        Napi::Object jsStatsObject = info[0].ToObject();
```

Comment:
`ToObject` will coerce to an object even if it's not which isn't what you want.

```suggestion
        Napi::Object jsStatsObject = info[0].As<Napi::Object>();
```